### ISSUE TITLE
Check redundancy setup with paramiko 2.2.0

### ISSUE THREAD

**Issue by fredreichbier** [NetKnights-GmbH/privacyidea-appliance#20]
The following snippet models the behavior of our redundancy setup: We connect to a SSH server twice using the same ``SSHClient`` object. This works without problems for paramiko 1.16.0 (which is shipped with Ubuntu 16.04), but fails with paramiko 2.2.0:

```python
import paramiko
client = paramiko.SSHClient()
client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
client.connect('192.168.33.151', username='root', password='test')
client.close()
client.connect('192.168.33.151', username='root', password='test')
```

Error on paramiko 2.2.0:
```
Traceback (most recent call last):
  File "test.py", line 6, in <module>
    client.connect('192.168.33.151', username='root', password='test')
  File "/tmp/par/paramiko1/lib/python2.7/site-packages/paramiko/client.py", line 380, in connect
    raise BadHostKeyException(hostname, server_key, our_key)
paramiko.ssh_exception.BadHostKeyException: ('192.168.33.151', <paramiko.ed25519key.Ed25519Key object at 0x7f7aa2e54cd0>, <paramiko.ed25519key.Ed25519Key object at 0x7f7aa2e54610>)
```

**Comment by fredreichbier** [comment]
I think this gets interesting once distributions start shipping paramiko 2.2.0.

**Comment by cornelinux** [comment]
Hi @fredreichbier,
we should update paramiko probably due to security issues.
Can you please check with ubuntu16 and ubuntu18 and update the dependency accordingly.

**Comment by fredreichbier** [comment]
OK, so we need to check that the appliance tool works for
* paramiko 1.16 (shipped by Ubuntu 16.04)
* paramiko 2.0.0 (shipped by Ubuntu 18.04)
* paramiko 2.4.2 (current stable)

**Comment by cornelinux** [comment]
If we would ship the appliance in a virtualenv just like privacyidea itself, we could also tag to a certain version of paramiko.

**Comment by fredreichbier** [comment]
I just re-checked the redundancy setup for the different versions of paramiko:
* paramiko 1.16.0: redundancy setup works both with a tinc tunnel and without a tinc tunnel
* paramiko 2.0.0: redundancy setup works both with a tinc tunnel and without a tinc tunnel
* paramiko 2.2.0: redundancy setup works *with* a tinc tunnel, but fails with a ``paramiko.ssh_exception.BadHostKeyException`` *without* a tinc tunnel, like described above. Setup with a tinc tunnel works because in this cas,e we do not actually use the same paramiko SSH client to connect to the same IP twice: First, we connect to the actual IP, and then we use the VPN IP.
* paramiko 2.4.2: redundancy setup works with a tinc tunnel and without a tinc tunnel.

In conclusion: If we use the paramiko versions 1.16.0 or 2.0.0 provided by Ubuntu 16.04 or 18.04, redundancy setup works. If we ship the appliance in a virtualenv, we can pin paramiko to the latest stable 2.4.2.

**Comment by plettich** [comment]
Bumping paramiko to a version `>= 2.0.0` would also avoid installing the deprecated `pycrypto` package. paramiko `>= 2.0.0` uses `cryptography`.
